packages/typescript-vfs/src/index.ts

```diff
@@ -99,6 +100,13 @@ export function createVirtualTypeScriptEnvironment(
 
       updateFile(newSourceFile)
     },
+    deleteFile(fileName) {
+      const sourceFile = languageService.getProgram()!.getSourceFile(fileName)
+      if (!sourceFile) {
+        throw new Error("Did not find a source file for " + fileName)
+      }
+      deleteFile(sourceFile)
```

Comment:
Not sure either way, but does it make sense to throw on delete of a file that doesn't exist? Or should this just be a noop? The canonical implementation in TS itself catches errors, so never observes Node's throws for a nonexistent file.

Reply:
I don't have a strong opinion either way. For updateFile it make sense but a nooop here is probably acceptable

Reply:
Yeah, I would probably make this operation not throw.

Reply:
Removed 👍

Reply:
Thanks!